src/scripts/main.js

```diff
@@ -1 +1,120 @@
 'use strict';
+
+window.addEventListener('DOMContentLoaded', () => {
+  class Burger {
+    constructor(btn, list) {
+      this.btn = btn;
+      this.list = list;
+      this.opened = false;
+      this.btnDisabled = false;
+      this.scrollWidth = window.innerWidth - document.body.clientWidth;
+
+      this.btn.addEventListener('click', () => {
+        this.toggle();
+      });
+    }
+
+    open() {
+      this.disableBtn();
+
+      if (this.scrollWidth) {
+        document.body.style.paddingRight = `${this.scrollWidth}px`;
+
+        document.querySelectorAll('[data-fixed-padding]').forEach((el) => {
+          el.style.paddingRight = `${this.scrollWidth}px`;
+        });
+      }
+
+      this.btn.classList.add('burger--active');
+      this.btn.setAttribute('aria-expanded', true);
+      this.list.classList.add('nav--opened');
+      this.list.setAttribute('aria-hidden', false);
+
+      document.body.classList.add('page__content--lock');
+    }
+
+    close() {
+      this.disableBtn();
+
+      if (this.scrollWidth) {
+        document.body.style.paddingRight = null;
+
+        document.querySelectorAll('[data-fixed-padding]').forEach((el) => {
+          el.style.paddingRight = null;
+        });
+      }
+
+      document.body.classList.remove('page__content--lock');
+      this.btn.classList.remove('burger--active');
+      this.btn.setAttribute('aria-expanded', false);
+      this.list.classList.remove('nav--opened');
+      this.list.setAttribute('aria-hidden', true);
+    }
+
+    toggle() {
+      if (this.btn.classList.contains('burger--active')) {
+        this.close();
+      } else {
+        this.open();
+      }
+    }
+
+    disableBtn() {
+      this.btn.disabled = true;
+
+      this.list.addEventListener('transitionend', () => {
+        this.btn.disabled = false;
+      });
+    }
+  }
+
+  class Debounce {
+    constructor(time = 100, functions = () => { }) {
+      this.time = time;
+      this.functions = functions;
+      this.timeOut = null;
+    }
+
+    init() {
+      clearTimeout(this.timeOut);
+
+      this.timeOut = setTimeout(() => {
+        this.functions();
+      }, this.time);
+    }
+  }
```

Comment:
if you know how to implement debounce decorator js will be easy for you 🙂
if you will implement throttle without googling I will buy you a chocolate

Reply:
I already know about throttle, watching some videos before. In this case, we don't need to periodically execute the function, so I decide to implement debounce decorator. But to be honest, I can't implement them without googling a little bit.